### PR TITLE
:recycle: Refactor M_StartKill to receive a Monster reference

### PR DESCRIPTION
**Pull request by canassa**

### CODE REVIEW

File: Source/monster.cpp [hunk 1]
```diff
@@ -3970,15 +3970,12 @@ void StartMonsterDeath(Monster &monster, int pnum, bool sendmsg)
...
 {
-	assert(monsterId >= 0 && monsterId < MaxMonsters);
-	Monster &monster = Monsters[monsterId];
-
 	if (pnum == MyPlayerId) {
-		delta_kill_monster(monsterId, monster.position.tile, *MyPlayer);
-		if (monsterId != pnum) {
-			NetSendCmdLocParam1(false, CMD_MONSTDEATH, monster.position.tile, monsterId);
+		delta_kill_monster(monster.getId(), monster.position.tile, *MyPlayer);
+		if (monster.getId() != pnum) {
```
Comment (canassa) [inline Source/monster.cpp:3977]: This introduces the following warning:

```
warning: comparison of integers of different signs: 'size_t' (aka 'unsigned long') and 'int' [-Wsign-compare]
```

But I am not sure how to *properly* solve it. Monster::getId returns a size_t but the playerId are int

Comment (glebm) [reply Source/monster.cpp:3977]: `static_cast<size_t>(pnum)`

Comment (canassa) [reply Source/monster.cpp:3977]: But that's just hidding the issue, no?

Comment (glebm) [reply Source/monster.cpp:3977]: It makes it explicit that the comparison is intended

Comment (canassa) [reply Source/monster.cpp]: Thanks! I updated it

File: Source/monster.cpp [hunk 2]
```diff
@@ -3987,15 +3988,12 @@ void StartMonsterDeath(Monster &monster, int pnum, bool sendmsg)
...
 {
-	assert(static_cast<size_t>(monsterId) < MaxMonsters);
-	Monster &monster = Monsters[monsterId];
-
 	if (pnum == MyPlayerId) {
-		delta_kill_monster(monsterId, monster.position.tile, *MyPlayer);
-		if (monsterId != pnum) {
-			NetSendCmdLocParam1(false, CMD_MONSTDEATH, monster.position.tile, monsterId);
+		delta_kill_monster(monster.getId(), monster.position.tile, *MyPlayer);
+		if (monster.getId() != static_cast<size_t>(pnum)) {
```
Comment (AJenbo) [inline Source/monster.cpp:3995]:
```suggestion
		if (&monster != &Monsters[pnum]) {
```

File: Source/monster.cpp [hunk 3]
```diff
@@ -3987,15 +3988,12 @@ void StartMonsterDeath(Monster &monster, int pnum, bool sendmsg)
...
+void M_StartKill(Monster &monster, int pnum)
 {
-	assert(static_cast<size_t>(monsterId) < MaxMonsters);
-	Monster &monster = Monsters[monsterId];
-
 	if (pnum == MyPlayerId) {
-		delta_kill_monster(monsterId, monster.position.tile, *MyPlayer);
-		if (monsterId != pnum) {
-			NetSendCmdLocParam1(false, CMD_MONSTDEATH, monster.position.tile, monsterId);
+		delta_kill_monster(monster.getId(), monster.position.tile, *MyPlayer);
```
Comment (AJenbo) [inline Source/monster.cpp:3994]: Should cleanup the leafs first so we don't end up with a lot of ugly calls to `.getId()`

Comment (canassa) [reply Source/monster.cpp:3994]: For some reason, I thought that `delta_kill_monster` was going to remain the same. I will update it.

Comment (AJenbo) [reply Source/monster.cpp:3994]: it is monsterid, the golems have the same id as there player, which is how you look them up.

Comment (canassa) [reply Source/monster.cpp:3994]: Opened #4988

Comment (canassa) [reply Source/monster.cpp]: Alright, it's rebased!

File: Source/missiles.cpp [hunk 4]
```diff
@@ -2156,13 +2156,14 @@ void AddGolem(Missile &missile, const AddMissileParameter &parameter)
 
 	int playerId = missile._misource;
 	Player &player = Players[playerId];
+	Monster &monster = Monsters[playerId];
```
Comment (AJenbo) [inline Source/missiles.cpp:2159]: Since we know more specificially
```suggestion
	Monster &golem = Monsters[playerId];
```

Comment (canassa) [reply Source/missiles.cpp]: Done

**Review by AJenbo — APPROVED** [review]
Nice :)

**Pull request merged**